Bottom: addfa57f1218d7719b0785fd5ad73ffd33f6d5ca
Top:    af22931162739b6ef27bde9ea60cb16cfbb13900
Author: Mark de Wever <koraq@xs4all.nl>
Date:   2021-08-07 10:04:37 +0200

Add an initial GUI.

Just an intial experiment to have some kind of GUI for the application.


---

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2f61c76..51058b6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,4 +1,5 @@
 add_subdirectory(modules)
+
 add_executable(rpn
 	rpn.cpp
 )
diff --git a/src/modules/calculator/CMakeLists.txt b/src/modules/calculator/CMakeLists.txt
index e3e7585..16ce5a7 100644
--- a/src/modules/calculator/CMakeLists.txt
+++ b/src/modules/calculator/CMakeLists.txt
@@ -11,6 +11,7 @@ target_link_libraries(calculator
 	PRIVATE
 		math
 )
+add_module(calculator calculator.cpp)
 add_module(calculator.controller controller.cpp)
 add_module(calculator.model model.cpp)
 add_module(calculator.value value.cpp)
@@ -20,6 +21,11 @@ add_dependencies(calculator.model.pcm
 add_dependencies(calculator.controller.pcm
 	calculator.model.pcm
 )
-add_dependencies(calculator
+add_dependencies(calculator.pcm
+	calculator.value.pcm
+	calculator.model.pcm
 	calculator.controller.pcm
 )
+add_dependencies(calculator
+	calculator.pcm
+)
diff --git a/src/modules/calculator/model.cpp b/src/modules/calculator/model.cpp
index c799329..79e7f42 100644
--- a/src/modules/calculator/model.cpp
+++ b/src/modules/calculator/model.cpp
@@ -18,7 +18,7 @@ export import calculator.value;
 
 import<string>;
 import<string_view>;
-import<vector>;
+export import<vector>;
 
 namespace calculator {
